types/datum.go

```diff
@@ -798,6 +798,9 @@ func (d *Datum) convertToString(sc *stmtctx.StatementContext, target *FieldType)
 		s = strconv.FormatFloat(d.GetFloat64(), 'f', -1, 64)
 	case KindString, KindBytes:
 		s = d.GetString()
+		if target.Tp == mysql.TypeString {
+			s = strings.TrimRight(s, " ")
```

Comment:
Should we consider other blank characters here? `\n`, `\t`,etc

Reply:
I did not find specific trailing spaces in "https://dev.mysql.com/doc/refman/8.0/en/char.html".
I tried the following situations and the results are the same.
```
insert into vctt values ("ab  ", " a\a\a\a\a");
insert into vctt values ("ab  ", " a\b\b\b\b");
insert into vctt values ("ab  ", " a\f\f\f\f");
insert into vctt values ("ab  ", " a\v\v\v\v");

insert into vctt values ("ab  ", " a\n\n\n\n");
insert into vctt values ("ab  ", " a\r\r\r\r");
insert into vctt values ("ab  ", " a\t\t\t\t");
insert into vctt values ("ab  ", " a        ");

insert into vctt values ("ab  ", " a\\\\\\\\");
insert into vctt values ("ab  ", " a\'\'\'\'");
insert into vctt values ("ab  ", " a\"\"\"\"");
insert into vctt values ("ab  ", " a\?\?\?\?");
insert into vctt values ("ab  ", " a\0\0\0\0");
```